packages/ag-charts-community/src/util/types.ts

```diff
@@ -1 +1,19 @@
 export type Has<P extends keyof T, T> = T & { [K in P]-?: T[P] };
+
+export type Mutable<T> = T extends object ? { -readonly [K in keyof T]: Mutable<T[K]> } : T;
+
+export type Defined<T> = T extends undefined ? never : T;
+
+export type DeepRequired<T> = T extends (...args: any[]) => any
```

Comment:
Side-note: I'm a little skeptical about how well this will work, TS (3.x from memory) has historically had major issues with cyclic types like this - but maybe they have addressed this in more recent versions?